src/transaction/log_recovery.c

```diff
@@ -2809,12 +2809,13 @@ log_recovery_redo (THREAD_ENTRY * thread_p, const LOG_LSA * start_redolsa, const
 			    {
 			      temp_length -= LOGAREA_SIZE - (int) (log_lsa.offset);
 			      assert (log_pgptr != NULL);
-			      if ((logpb_fetch_page (thread_p, ++log_lsa.pageid, log_pgptr)) == NULL)
+			      if ((logpb_fetch_page (thread_p, &log_lsa, LOG_CS_FORCE_USE, log_pgptr)) == NULL)
```

Comment:
@razvanradulescu 

I don't think to fetch a pge by `log_lsa` is correct. 
The code block fetches the next page(s), since the log record does not fit in the page that `log_lsa` points to. 
Please notice that the existing code increments pageid and then fetch the page.